src/htm/encoders/ScalarEncoder.cpp

```diff
@@ -40,8 +40,8 @@ void ScalarEncoder::initialize(const ScalarEncoderParameters &parameters)
   if( parameters.sparsity   > 0.0f) { num_active_args++; }
   NTA_CHECK( num_active_args != 0u )
       << "Missing argument, need one of: 'activeBits' or 'sparsity'.";
-  NTA_CHECK( num_active_args == 1u )
-      << "Too many arguments, choose only one of: 'activeBits' or 'sparsity'.";
+  if( num_active_args > 1u )
+    NTA_WARN  << "Specified both: 'activeBits' and 'sparsity'. Sparsity takes precedence. Or specify only one of them.";
```

Comment:
This used to assert and fail on incompatible (too many specified) params. I changed it now to only warn and select one of the params (all are equivalent). 

Because we internally compute all (other) params, and serialization stores all of them (there's no way to know which were orig and which were re-computed). Alternative would be serializing only a (hard-coded) subset of the params?

Reply:
> lternative would be serializing only a (hard-coded) subset of the params?

maybe that is a better approach? As now I can specify not equivalent params (activebits=100, sparsity=0.001) and silently (with warning) one takes preference (sparsity)